### DISCUSSION TITLE
Tasmota ESP32 Serial Bridge

### DISCUSSION THREAD

**Discussion by DimaMukhin84** [arendst/Tasmota#22312] [Q&A]
Good afternoon.
I have ESP32 with Tasmota firmware
I need to send UART data to MQTT and receive.
I have connected ESP32 UART-USB adapter.
I send data to the port (Hello) and they are displayed in the MQTT broker.
tele>
     tasmota_A9DD5C >
                                  RESULT={“SSerialReceived”: “Hello”}

Which topic should I go to and how do I write a message to get the ESP32 to output them to the UART port ?

![image](https://github.com/user-attachments/assets/e1273596-0b61-4b50-8c31-df0ac09d548c)

![image](https://github.com/user-attachments/assets/a7edb4b3-b63e-45dc-ba41-0a6267c2fa68)

**Comment by sfromis** [comment]
Tasmota commands can be published to the `cmnd` topic prefix (instead of you seeing `tele` on incoming messages) , and you can see the options for send commands here: https://tasmota.github.io/docs/Commands/#serial-bridge
Remember to have the command argument in the separate payload field.

**Comment by DimaMukhin84** [comment]
I've read all that, but it doesn't help.
ChatGTP helped.
Here's the top from which Tasmota reads messages

cmnd/tasmota_A9DD5C/SSerialSend
